### PR TITLE
feat: `library(duckplyr)` calls `methods_overwrite()`

### PR DESCRIPTION
**Pull request by krlmlr**

### CODE REVIEW

**Comment by krlmlr** [comment]
@DavisVaughan: Do you see issues with this way of overwriting and talking about it? Is there value in rearranging the two howto sections in the README?

File: README.Rmd [hunk 1]
```diff
@@ -69,14 +69,14 @@ pak::pak("duckdblabs/duckplyr")
 
 ```{r attach}
 library(conflicted)
-library(duckplyr)
+library(dplyr)
 conflict_prefer("filter", "duckplyr")
```
Comment (DavisVaughan) [inline README.Rmd:73]: do you still want this particular conflict_prefer call?

File: README.Rmd [hunk 2]
```diff
@@ -69,14 +69,14 @@ pak::pak("duckdblabs/duckplyr")
...
 ```
 
 There are two ways to use duckplyr.
 
-1. To enable duckplyr for individual data frames, use `as_duckplyr_df()` as the first step in your pipe.
-1. To enable duckplyr for the entire session, use `methods_overwrite()`.
+1. To enable duckplyr for individual data frames, use `duckplyr::as_duckplyr_df()` as the first step in your pipe, without attaching the package.
+1. By calling `library(duckplyr)`, it is enabled for the entire session. To turn this off, call `methods_restore()`.
```
Comment (DavisVaughan) [inline README.Rmd:79]:
```suggestion
1. By calling `library(duckplyr)`, it overwrites dplyr methods and is automatically enabled for the entire session without having to call `as_duckplyr_df()`. To turn this off, call `methods_restore()`.
```

I think we need to be very clear about the fact that this _overwrites_ (that's the key word to me)

File: R/zzz-methods.R [hunk 3]
```diff
@@ -26,7 +26,12 @@ on_load({
 #' tibble(a = 1:3) %>%
 #'   mutate(b = a + 1)
 methods_overwrite <- function() {
-  cli::cli_inform(c(i = "Overwriting {.pkg dplyr} methods with {.pkg duckplyr} methods"))
+  cli::cli_inform(
+    c(
+      v = "Overwriting {.pkg dplyr} methods with {.pkg duckplyr} methods",
+      i = "Turn off with {.run duckplyr::methods_restore()}"
```
Comment (DavisVaughan) [inline R/zzz-methods.R:32]:
```suggestion
      v = "Overwriting {.pkg dplyr} methods with {.pkg duckplyr} methods.",
      i = "Turn off with {.run duckplyr::methods_restore()}."
```

We do at least make an attempt at using full stops at the end of all our bullets

**Review by DavisVaughan — APPROVED** [review]
Seems good like this, I think the readme ordering is right (less to more aggressive in terms of how to use the package).


What happens if you `library(duckplyr)` and _then_ `library(dplyr)`? Maybe that's not a problem since we Import dplyr, and it always at least gets _loaded_ first?

File: README.Rmd (hunk 1 above)
Comment (krlmlr) [reply README.Rmd:73]:
```suggestion
conflict_prefer("filter", "dplyr")
```

Comment (krlmlr) [reply README.Rmd:73]: Good catch, thanks!

File: README.Rmd [hunk 4]
```diff
@@ -131,10 +131,10 @@ out
 
 This example illustrates usage of duckplyr for all data frames in the R session.
 
-Use `methods_overwrite()` to enable processing with duckdb for all data frames:
+Use `library(duckplyr)` or `duckplyr::methods_overwrite()` to enable processing with duckdb for all data frames:
```
Comment (krlmlr) [inline README.Rmd:134]:
```suggestion
Use `library(duckplyr)` or `duckplyr::methods_overwrite()` to overwrite dplyr methods and enable processing with duckdb for all data frames:
```

**Pull request merged**